system/include/emscripten/html5_webgl.h

```diff
@@ -14,7 +14,7 @@
 extern "C" {
 #endif
 
-typedef intptr_t EMSCRIPTEN_WEBGL_CONTEXT_HANDLE;
+typedef uintptr_t EMSCRIPTEN_WEBGL_CONTEXT_HANDLE;
```

Comment:
just a thought: As far as I can tell this is always a real heap pointer. Would it make sense to make it a real (opaque) pointer type? Like:
`typedef struct EMSCRIPTEN_WEBGL_CONTEXT_S* EMSCRIPTEN_WEBGL_CONTEXT_HANDLE;`
But I this would be likely to break more code, so I guess it wouldn't.

Reply:
Actually, its only a pointer in some configurations.  In the default configuration is a handle allocated by our `HandleAllocator` which normally we map to an `int`.

Having said that we could certainly use an opaque pointer type here if we want..  I think this change might be slightly safer since in terms of not breaking existing code.

Reply:
See https://github.com/emscripten-core/emscripten/blob/bbbbdb138f77b3e16c751e0e19a48cb4d16be02c/src/library_webgl.js#L1055-L1069

Reply:
Ah, I looked there, but missed the `GL.getNewId` among the preprocessor stuff. But true we could use a pointer anyway.

I agree let's keep the uintptr_t for better compatibility.